src/vs/workbench/browser/parts/paneCompositeBar.ts

```diff
@@ -565,19 +565,21 @@ export class PaneCompositeBar extends Disposable {
 	}
 
 	private storeCachedViewContainersState(cachedViewContainers: ICachedViewContainer[]): void {
-		this.setPinnedViewContainers(cachedViewContainers.map(({ id, pinned, visible, order }) => (<IPinnedViewContainer>{
+		const pinnedViewContainers = this.getPinnedViewContainers();
+		this.setPinnedViewContainers(cachedViewContainers.map(({ id, pinned, order }) => (<IPinnedViewContainer>{
 			id,
 			pinned,
-			visible,
+			visible: pinnedViewContainers.find(({ id }) => id === id)?.visible,
```

Comment:
Could the identifier be renamed to reduce confusion?

Reply:
Good catch. Thanks. Done